topology/enhancers/topology_manager.go

```diff
@@ -119,12 +120,11 @@ func (tm *TopologyManager) createEdge(edge *types.EdgeRule) error {
 		if !topology.HaveOwnershipLink(tm.graph, src[0], dst[0]) {
 			topology.AddOwnershipLink(tm.graph, src[0], dst[0], nil)
 		}
-	case "both":
-		if !topology.HaveLayer2Link(tm.graph, src[0], dst[0]) {
-			topology.AddLayer2Link(tm.graph, src[0], dst[0], edge.Metadata)
-		}
-		if !topology.HaveOwnershipLink(tm.graph, src[0], dst[0]) {
-			topology.AddOwnershipLink(tm.graph, src[0], dst[0], nil)
+	default:
+		id, _ := uuid.NewV5(uuid.NamespaceOID, []byte(string(src[0].ID)+string(dst[0].ID)+edge.Metadata["RelationType"].(string)))
+		_, err := tm.graph.NewEdge(graph.Identifier(id.String()), src[0], dst[0], edge.Metadata)
```

Comment:
We should probably check if a link with the same relationtype already exist

Reply:
Done

Reply:
You could simply use the GenID function giving the src + dst + relationtype, it will do the UUID for you
https://github.com/skydive-project/skydive/blob/master/graffiti/graph/graph.go#L310